src/main/java/ua/lviv/iot/parkingServer/logic/ParkingService.kt

```diff
@@ -0,0 +1,43 @@
+package ua.lviv.iot.parkingServer.logic
+
+import org.springframework.stereotype.Service
+import ua.lviv.iot.parkingServer.datastorage.ParkingFileStore
+import ua.lviv.iot.parkingServer.model.Parking
+import java.util.concurrent.atomic.AtomicLong
+import javax.annotation.PostConstruct
+import javax.annotation.PreDestroy
+
+@Service
+class ParkingService(private val parkingFileStore: ParkingFileStore) {
+    private val parking: MutableMap<Long, Parking> = HashMap()
+    private var id: AtomicLong = AtomicLong(1L)
+
+    fun findAllParking(): List<Parking> = ArrayList<Parking>(parking.values)
+
+    fun findParkingById(id: Long): Parking? = parking[id]
```

Comment:
The return type must be non-nullable.

When you request an entity by id then there are two possible results:
1. actual entity with the requested ID
2. exception if entity not found
     - at service layer we usually throw exception
     - at controller layer we have an exception handler or controller advice that converts the exception to 404 response
     
     
Let's just throw here for now:
```kotlin
fun findParkingById(id: Long): Parking = parking[id] ?: EntityNotFoundException("Parking with id=$id not found")

class EntityNotFoundException(msg: String) : RuntimeException(msg)
```

----

Please update the controller's related function to non-nullable return type.
Also, review another services and controllers with similar logic

Reply:
✅